packages/kbn-zod-helpers/kibana.jsonc

```diff
@@ -1,6 +1,10 @@
 {
-  "devOnly": false,
+  "type": "shared-common",
```

Comment:
What's the difference between the newly added properties and this existing `type` property?

Reply:
They cover 2 different dimensions of categorisation:

* `type` categorises the modules in either browser-side, server-side or shared across browser and server.
* `group` categorises the modules in either platform (used by all solutions) or one of the solution buckets. **This is the categorisation we're performing with this PR**.